aws/resource_aws_emr_instance_group.go

```diff
@@ -1,32 +1,73 @@
 package aws
 
 import (
+	"encoding/json"
 	"errors"
+	"fmt"
 	"log"
+	"strings"
 	"time"
 
-	"fmt"
-
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/emr"
 	"github.com/hashicorp/terraform/helper/resource"
 	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/structure"
+	"github.com/hashicorp/terraform/helper/validation"
 )
 
-var emrInstanceGroupNotFound = errors.New("No matching EMR Instance Group")
+var errEMRInstanceGroupNotFound = errors.New("No matching EMR Instance Group")
```

Comment:
Nit: We can use the upstream [`resource.NotFoundError`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#NotFoundError) and `isResourceNotFoundError()` helper instead of our own custom error here.